nixos/modules/services/monitoring/prometheus/alertmanager.nix

```diff
@@ -0,0 +1,117 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alertmanager;
+  mkConfigFile = pkgs.writeText "alertmanager.yml" cfg.configuration;
+in {
+  options = {
+    services.alertmanager = {
+      enable = mkEnableOption "Prometheus Alertmanager";
+
+      user = mkOption {
+        type = types.str;
+        default = "root";
+        description = ''
+          User name under which Alertmanager shall be run.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "root";
+        description = ''
+          Group under which Alertmanager shall be run.
+        '';
+      };
+
+      configuration = mkOption {
+        type = types.str;
```

Comment:
You could also use a nix attrset and write that as JSON (JSON is a subset of YAML)